File: C/Algorithms/1539.c
int findKthPositive(int *arr, int arrSize, int k) {
  int start = 1;
  for (int i = 0; i < arrSize; i++) {
    int j = *(arr + i);
    if (start == j)
      start++;
    else {
      int step = j - start;
      if (step >= k)
        return start + k - 1;
      k -= step;
      start = j + 1;
    }
  }
  return start + k - 1;
}

File: Rust/Algorithms/1539.rs
impl Solution {
    pub fn find_kth_positive(arr: Vec<i32>, mut k: i32) -> i32 {
        let mut start = 1;
        for j in arr.into_iter() {
            if (start == j) {
                start += 1;
            } else {
                let step = j - start;
                if (step >= k) {
                    return start + k - 1;
                }
                k -= step;
                start = j + 1;
            }
        }
        start + k - 1
    }
}
